cmd/istio-cni-repair/main.go

```diff
@@ -0,0 +1,230 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A simple daemonset binary to repair pods that are crashlooping
+// after winning a race condition against istio-cni
+package main
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	client "k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"istio.io/cni/pkg/repair"
+	"istio.io/pkg/log"
+)
+
+type ControllerOptions struct {
+	DaemonPollPeriod int             `json:"daemon_poll_period"`
+	RepairOptions    *repair.Options `json:"repair_options"`
+	DeletePods       bool            `json:"delete_pods"`
+	LabelPods        bool            `json:"label_pods"`
+	CreateEvents     bool            `json:"create_events"`
+	RunAsDaemon      bool            `json:"run_as_daemon"`
+}
+
+var (
+	loggingOptions = log.DefaultOptions()
+)
+
+// Parse command line options
+func parseFlags() (filters *repair.Filters, options *ControllerOptions) {
+	// Parse command line flags
+	// Filter Options
+	pflag.String("node-name", "", "The name of the node we are managing (will manage all nodes if unset)")
+	pflag.String(
+		"sidecar-annotation",
+		"sidecar.istio.io/status",
+		"An annotation key that indicates this pod contains an istio sidecar. All pods without this annotation will be ignored."+
+			"The value of the annotation is ignored.")
+	pflag.String(
+		"init-container-name",
+		"istio-init",
+		"The name of the istio init container (will crash-loop if CNI is not configured for the pod)")
+	pflag.String(
+		"init-container-termination-message",
+		"",
+		"The expected termination message for the init container when crash-looping because of CNI misconfiguration")
+	pflag.Int(
+		"init-container-exit-code",
+		126,
+		"Expected exit code for the init container when crash-looping because of CNI misconfiguration")
+
+	pflag.String("label-selectors", "", "A set of label selectors in label=value format that will be added to the pod list filters")
+	pflag.String("field-selectors", "", "A set of field selectors in label=value format that will be added to the pod list filters")
+
+	// Repair Options
+	pflag.Bool("delete-pods", false, "Controller will delete pods")
+	pflag.Bool("label-pods", false, "Controller will label pods")
+	pflag.Bool("create-events", true, "Controller will create Kubernetes events")
+	pflag.Bool("run-as-daemon", false, "Controller will run in a loop")
+	pflag.Int("daemon-poll-period", 10, "Polling period for daemon (in seconds)")
+	pflag.String(
+		"broken-pod-label-key",
+		"cni.istio.io/uninitialized",
+		"The key portion of the label which will be set by the reconciler if --label-pods is true")
+	pflag.String(
+		"broken-pod-label-value",
+		"true",
+		"The value portion of the label which will be set by the reconciler if --label-pods is true")
+
+	pflag.Bool("help", false, "Print usage information")
```

Comment:
nit: Use [Cobra's built-in `help` command](https://github.com/spf13/cobra#help-command) like the rest of the Istio code, instead of implementing your own.